src/metabase/api/pulse.clj

```diff
@@ -0,0 +1,66 @@
+(ns metabase.api.pulse
+  "/api/pulse endpoints."
+  (:require [korma.core :refer [where subselect fields order limit]]
+            [compojure.core :refer [defroutes GET PUT POST DELETE]]
+            [hiccup.core :refer [html]]
+            [medley.core :refer :all]
+            [metabase.api.common :refer :all]
+            [metabase.db :as db]
+            [metabase.driver :as driver]
+            (metabase.models [card :refer [Card]]
+                             [database :refer [Database]]
+                             [hydrate :refer :all]
+                             [pulse :refer [Pulse] :as pulse])
+            [metabase.util :as util]
+            [metabase.pulse :as p]))
+
+
+(defendpoint GET "/"
+  "Fetch all `Pulses`"
+  []
+  (pulse/retrieve-pulses))
+
+
+(defendpoint POST "/"
+  "Create a new `Pulse`."
+  [:as {{:keys [name cards channels] :as body} :body}]
+  {name     [Required NonEmptyString]
+   cards    [Required ArrayOfMaps]
+   channels [Required ArrayOfMaps]}
+  (->500 (pulse/create-pulse name *current-user-id* (filter identity (map :id cards)) channels)))
+
+
+(defendpoint GET "/:id"
+  "Fetch `Pulse` with ID."
+  [id]
+  (->404 (pulse/retrieve-pulse id)))
+
+
+(defendpoint PUT "/:id"
+  "Update a `Pulse` with ID."
+  [id :as {{:keys [name cards channels] :as body} :body}]
+  {name     [Required NonEmptyString]
+   cards    [Required ArrayOfMaps]
+   channels [Required ArrayOfMaps]}
+  (check-404 (db/exists? Pulse :id id))
+  (pulse/update-pulse {:id       id
+                       :name     name
+                       :cards    (filter identity (into [] (map :id cards)))
```

Comment:
`filter` returns a lazy seq so I think the `into []` won't have any effect. If you want a vector though you can use `filterv`